bundles/org.openhab.binding.intesis/src/main/java/org/openhab/binding/intesis/internal/api/IntesisBoxSocketApi.java

```diff
@@ -0,0 +1,232 @@
+        tcpInput = new BufferedReader(new InputStreamReader(tcpSocket.socket.getInputStream()));
+
+        Thread tcpListener = new Thread(new TCPListener());
+        tcpListener.start();
+
+        setConnected(true);
+        if (listener != null) {
+            listener.connectionStatusChanged(ThingStatus.ONLINE);
+        }
+    }
+
+    @SuppressWarnings("null")
+    public void closeConnection() {
+        try {
+            if (tcpSocket != null) {
+                logger.debug("closeConnection(): Closing Socket!");
+                tcpSocket.close();
+                tcpSocket = null;
+            }
+            if (tcpInput != null) {
+                logger.debug("closeConnection(): Closing Output Writer!");
+                tcpInput.close();
+                tcpInput = null;
+            }
+            if (tcpOutput != null) {
+                logger.debug("closeConnection(): Closing Input Reader!");
+                tcpOutput.close();
+                tcpOutput = null;
+            }
+
+            setConnected(false);
+            logger.debug("closeConnection(): Closed TCP Connection!");
+        } catch (IOException ioException) {
+            logger.debug("closeConnection(): Unable to close connection - {}", ioException.getMessage());
+        } catch (Exception exception) {
+            logger.debug("closeConnection(): Error closing connection - {}", exception.getMessage());
+        }
+    }
+
+    private class TCPListener implements Runnable {
+        private final Logger logger = LoggerFactory.getLogger(TCPListener.class);
+
+        /**
+         * Run method. Runs the MessageListener thread
+         */
+        @Override
+        public void run() {
+            try {
+                while (isConnected()) {
+                    String message = read();
+                    try {
+                        logger.trace("Calling readMessage()");
+                        readMessage(message);
+                    } catch (Exception e) {
+                        logger.debug("TCPListener(): Message not handled by thing: {}", e.getMessage());
+                        closeConnection();
+                    }
+                }
+            } catch (Exception e) {
+                logger.debug("TCPListener(): Unable to read message: {} ", e.getMessage(), e);
+                closeConnection();
+            }
+        }
+    }
+
+    public void addIntesisBoxChangeListener(IntesisBoxChangeListener listener) {
+        if (this.changeListener == null) {
+            this.changeListener = listener;
+        }
+    }
+
+    private void write(String data) {
+        try {
+            if (tcpOutput != null) {
+                tcpOutput.write(data);
+            }
+            if (tcpOutput != null) {
+                tcpOutput.flush();
+            }
+        } catch (IOException ioException) {
+            logger.debug("write(): {}", ioException.getMessage());
```

Comment:
Can you update the Thing status with the exception's message? Same for the other occurrences.

Reply:
Done